src/pages/Top.js

```diff
@@ -9,10 +9,23 @@ const Top = () => {
   const [permission, setPermission] = useState('');
   const [accountData, setAccountData] = useState({});
 
+  const usePrevious = (value) => {
+    const ref = useRef();
+    useEffect(() => {
+      ref.current = value;
+    });
+    return ref.current;
+  }
+
+  const prevAmount = usePrevious({ error });
   useEffect(() => {
+    if(prevAmount && prevAmount.error) {
```

Comment:
We can just check for the truthiness of `prevError` if we invoke `usePrevious` with the specific `error` state instead of an object.